ethereum/contracts/query/QueryResponse.sol

```diff
@@ -434,6 +460,71 @@ abstract contract QueryResponse {
         checkLength(pcr.response, respIdx);
     }
 
+    /// @dev parseSolanaPdaQueryResponse parses a ParsedPerChainQueryResponse for a Solana Pda per-chain query.
+    function parseSolanaPdaQueryResponse(ParsedPerChainQueryResponse memory pcr) public pure returns (SolanaPdaQueryResponse memory r) {
+        if (pcr.queryType != QT_SOL_PDA) {
+            revert UnsupportedQueryType();
```

Comment:
It would be nice to change this custom error to log the `expected` and `actual` query type.